File: posixlib/src/main/scala/scala/scalanative/posix/syslog.scala
package scala.scalanative
package posix

import scalanative.unsafe._

@extern
object syslog {
  @name("scalanative_closelog")
  def closelog(): Unit = extern

  @name("scalanative_openlog")
  def openlog(ident: CString, logopt: CInt, facility: CInt): Unit = extern

  @name("scalanative_setlogmask")
  def setlogmask(maskpri: CInt): CInt = extern

  @name("scalanative_log_emerg")
  def LOG_EMERG: CInt = extern

  @name("scalanative_log_alert")
  def LOG_ALERT: CInt = extern

  @name("scalanative_log_crit")
  def LOG_CRIT: CInt = extern

  @name("scalanative_log_err")
  def LOG_ERR: CInt = extern

  @name("scalanative_log_warning")
  def LOG_WARNING: CInt = extern

  @name("scalanative_log_notice")
  def LOG_NOTICE: CInt = extern

  @name("scalanative_log_info")
  def LOG_INFO: CInt = extern

  @name("scalanative_log_debug")
  def LOG_DEBUG: CInt = extern

  @name("scalanative_log_primask")
  def LOG_PRIMASK: CInt = extern

  @name("scalanative_log_pri")
  def LOG_PRI(p: CInt): CInt = extern

  @name("scalanative_log_makepri")
  def LOG_MAKEPRI(fac: CInt, pri: CInt): CInt = extern

  @name("scalanative_log_kern")
  def LOG_KERN: CInt = extern

  @name("scalanative_log_user")
  def LOG_USER: CInt = extern

  @name("scalanative_log_mail")
  def LOG_MAIL: CInt = extern

  @name("scalanative_log_daemon")
  def LOG_DAEMON: CInt = extern

  @name("scalanative_log_auth")
  def LOG_AUTH: CInt = extern

  @name("scalanative_log_syslog")
  def LOG_SYSLOG: CInt = extern

  @name("scalanative_log_lpr")
  def LOG_LPR: CInt = extern

  @name("scalanative_log_news")
  def LOG_NEWS: CInt = extern

  @name("scalanative_log_uucp")
  def LOG_UUCP: CInt = extern

  @name("scalanative_log_cron")
  def LOG_CRON: CInt = extern

  @name("scalanative_log_authpriv")
  def LOG_AUTHPRIV: CInt = extern

  @name("scalanative_log_ftp")
  def LOG_FTP: CInt = extern

  @name("scalanative_log_netinfo")
  def LOG_NETINFO: CInt = extern

  @name("scalanative_log_remoteauth")
  def LOG_REMOTEAUTH: CInt = extern

  @name("scalanative_log_install")
  def LOG_INSTALL: CInt = extern

  @name("scalanative_log_ras")
  def LOG_RAS: CInt = extern

  @name("scalanative_log_local0")
  def LOG_LOCAL0: CInt = extern

  @name("scalanative_log_local1")
  def LOG_LOCAL1: CInt = extern

  @name("scalanative_log_local2")
  def LOG_LOCAL2: CInt = extern

  @name("scalanative_log_local3")
  def LOG_LOCAL3: CInt = extern

  @name("scalanative_log_local4")
  def LOG_LOCAL4: CInt = extern

  @name("scalanative_log_local5")
  def LOG_LOCAL5: CInt = extern

  @name("scalanative_log_local6")
  def LOG_LOCAL6: CInt = extern

  @name("scalanative_log_local7")
  def LOG_LOCAL7: CInt = extern

  @name("scalanative_log_nfacilities")
  def LOG_NFACILITIES: CInt = extern

  @name("scalanative_log_facmask")
  def LOG_FACMASK: CInt = extern

  @name("scalanative_log_fac")
  def LOG_FAC(p: CInt): CInt = extern

  @name("scalanative_log_mask")
  def LOG_MASK(pri: CInt): CInt = extern

  @name("scalanative_log_upto")
  def LOG_UPTO(pri: CInt): CInt = extern

  @name("scalanative_log_pid")
  def LOG_PID: CInt = extern

  @name("scalanative_log_cons")
  def LOG_CONS: CInt = extern

  @name("scalanative_log_odelay")
  def LOG_ODELAY: CInt = extern

  @name("scalanative_log_ndelay")
  def LOG_NDELAY: CInt = extern

  @name("scalanative_log_nowait")
  def LOG_NOWAIT: CInt = extern

  @name("scalanative_log_perror")
  def LOG_PERROR: CInt = extern
}

File: posixlib/src/main/resources/scala-native/syslog.c
#include <syslog.h>
#include <stdarg.h>

void scalanative_closelog() { closelog(); }

void scalanative_openlog(const char *ident, int logopt, int facility) {
    openlog(ident, logopt, facility);
}

int scalanative_setlogmask(int maskpri) { return setlogmask(maskpri); }

int scalanative_log_emerg() { return LOG_EMERG; }
int scalanative_log_alert() { return LOG_ALERT; }
int scalanative_log_crit() { return LOG_CRIT; }
int scalanative_log_err() { return LOG_ERR; }
int scalanative_log_warning() { return LOG_WARNING; }
int scalanative_log_notice() { return LOG_NOTICE; }
int scalanative_log_info() { return LOG_INFO; }
int scalanative_log_debug() { return LOG_DEBUG; }
int scalanative_log_primask() { return LOG_PRIMASK; }

int scalanative_log_pri(int p) { return LOG_PRI(p); }
int scalanative_log_makepri(int fac, int pri) { return LOG_MAKEPRI(fac, pri); }

int scalanative_log_kern() { return LOG_KERN; }
int scalanative_log_user() { return LOG_USER; }
int scalanative_log_mail() { return LOG_MAIL; }
int scalanative_log_daemon() { return LOG_DAEMON; }
int scalanative_log_auth() { return LOG_AUTH; }
int scalanative_log_syslog() { return LOG_SYSLOG; }
int scalanative_log_lpr() { return LOG_LPR; }
int scalanative_log_news() { return LOG_NEWS; }
int scalanative_log_uucp() { return LOG_UUCP; }
int scalanative_log_cron() { return LOG_CRON; }
int scalanative_log_authpriv() { return LOG_AUTHPRIV; }
int scalanative_log_ftp() { return LOG_FTP; }

int scalanative_log_local0() { return LOG_LOCAL0; }
int scalanative_log_local1() { return LOG_LOCAL1; }
int scalanative_log_local2() { return LOG_LOCAL2; }
int scalanative_log_local3() { return LOG_LOCAL3; }
int scalanative_log_local4() { return LOG_LOCAL4; }
int scalanative_log_local5() { return LOG_LOCAL5; }
int scalanative_log_local6() { return LOG_LOCAL6; }
int scalanative_log_local7() { return LOG_LOCAL7; }

int scalanative_log_nfacilities() { return LOG_NFACILITIES; }
int scalanative_log_facmask() { return LOG_FACMASK; }

int scalanative_log_fac(int p) { return LOG_FAC(p); }

int scalanative_log_mask(int pri) { return LOG_MASK(pri); }
int scalanative_log_upto(int pri) { return LOG_UPTO(pri); }

int scalanative_log_pid() { return LOG_PID; }
int scalanative_log_cons() { return LOG_CONS; }
int scalanative_log_odelay() { return LOG_ODELAY; }
int scalanative_log_ndelay() { return LOG_NDELAY; }
int scalanative_log_nowait() { return LOG_NOWAIT; }
int scalanative_log_perror() { return LOG_PERROR; }
